pkg/resource/queue/sdk.go

```diff
@@ -103,6 +103,15 @@ func (rm *resourceManager) sdkFind(
 	ko.Spec.RedrivePolicy = resp.Attributes["RedrivePolicy"]
 	ko.Spec.VisibilityTimeout = resp.Attributes["VisibilityTimeout"]
 
+	// If the QueueName field is empty, populate it with the last part of the queue ARN
+	// This is a workaround for the fact that the QueueName field is required by the
+	// CreateQueue API call, but not by the GetQueueAttributes API call
+	// Use case: adopting an existing queue by queue URL
+	if ko.Spec.QueueName == nil {
+		split := strings.Split(string(tmpARN), ":")
```

Comment:
If I were to nitpick, you could use the built in `arn.Parse` as a safer method of getting the resource name.

Reply:
[refactor with arn.Parse](https://github.com/aws-controllers-k8s/sqs-controller/pull/32/commits/dc3ed46d682e52741e7b89670ee3c60998c1ab1c)